sql/hive/src/main/scala/org/apache/spark/sql/hive/TableReader.scala

```diff
@@ -288,29 +286,65 @@ class HadoopTableReader(
     }
   }
 
+  /**
+   * The entry of creating a RDD.
+   */
+  private def createHadoopRDD(
+      inputClassName: String, localTableDesc: TableDesc, inputPathStr: String): RDD[Writable] = {
+    if (classOf[org.apache.hadoop.mapreduce.InputFormat[_, _]]
+      .isAssignableFrom(Utils.classForName(inputClassName))) {
+      createNewHadoopRdd(localTableDesc, inputPathStr, inputClassName)
+    } else {
+      createOldHadoopRdd(localTableDesc, inputPathStr, inputClassName)
+    }
+  }
+
   /**
    * Creates a HadoopRDD based on the broadcasted HiveConf and other job properties that will be
    * applied locally on each slave.
    */
-  private def createHadoopRdd(
-    tableDesc: TableDesc,
-    path: String,
-    inputFormatClass: Class[InputFormat[Writable, Writable]]): RDD[Writable] = {
+  private def createOldHadoopRdd(
+      tableDesc: TableDesc, path: String, inputClassName: String): RDD[Writable] = {
 
     val initializeJobConfFunc = HadoopTableReader.initializeLocalJobConfFunc(path, tableDesc) _
 
     val rdd = new HadoopRDD(
       sparkSession.sparkContext,
       _broadcastedHadoopConf.asInstanceOf[Broadcast[SerializableConfiguration]],
       Some(initializeJobConfFunc),
-      inputFormatClass,
+      Utils.classForName(inputClassName)
+        .asInstanceOf[java.lang.Class[org.apache.hadoop.mapred.InputFormat[Writable, Writable]]],
```

Comment:
It's okay to make a variable for this if it hits line length limit and not sure about the style.

Reply:
I get it, done.